mobile/src/main/java/org/openhab/habdroid/model/Widget.kt

```diff
@@ -393,6 +414,10 @@ fun JSONObject.collectWidgets(parent: Widget?): List<Widget> {
         rawMinValue = optFloatOrNull("minValue"),
         rawMaxValue = optFloatOrNull("maxValue"),
         rawStep = optFloatOrNull("step"),
+        row = optInt("row"),
+        column = optInt("column"),
```

Comment:
You need to create the function `optIntOrNull()` and use it here. Copy the implementation from `optFloatOrNull()` ;)